src/scripts/convert_lovd2.php

```diff
@@ -0,0 +1,1318 @@
+    'vog' =>        array(
+        'output_header' =>          'Variants_On_Genome',
+        'customcol_prefix' =>       'VariantOnGenome',
+        'mandatory_fields' =>       array('id' => '0', 'allele' => '0', 'chromosome' => '0',
+            'position_g_start' => '0', 'position_g_end' => '0', 'type' => '?',
+            'VariantOnGenome/DNA' => 'g.?', 'statusid' => '', 'owned_by' => '0',
+            'created_by' => '0', 'created_date' => '0', 'edited_by' => '0', 'edited_date' => '0')),
+    'vot' =>        array(
+        'output_header' =>          'Variants_On_Transcripts',
+        'customcol_prefix' =>       'VariantOnTranscript',
+        'mandatory_fields' =>       array('id' => '0', 'transcriptid' => '0',
+            'position_c_start' => '0', 'position_c_start_intron' => '0', 'position_c_end' => '0',
+            'position_c_end_intron' => '0', 'VariantOnTranscript/Exon' => '?',
+            'VariantOnTranscript/RNA' => 'r.?')),
+    's2v' =>        array(
+        'output_header' =>          'Screenings_To_Variants',
+        'mandatory_fields' =>       array('screeningid' => '0', 'variantid' => '0')),
+);
+
+
+// Default user ID with which to overwrite user IDs in the input file. Used by
+// lovd_convertSubmitterID() and lovd_convertCuratorID().
+$sFixedSubmitterID = null;
+$sFixedCuratorID = null;
+
+// Translation array of LOVD2 user IDs to LOVD3 user IDs. Used by
+// lovd_convertSubmitterID() and lovd_convertCuratorID().
+$aSubmitterTranslationTable = array();
+$aCuratorTranslationTable = array();
+
+
+
+function lovd_autoIncIndividualID ($LOVD2PatientID)
+{
+    // ID generator for individuals.
+    return lovd_getInc('lovd_autoIncIndividualID');
+}
+
+
+
+
+
+function lovd_autoIncPhenotypeID ()
+{
+    // ID generator for phenotypes.
+    return lovd_getInc('lovd_autoIncPhenotypeID');
+}
+
+
+
+
+
+function lovd_autoIncVariantID ($LOVD2PatientID)
+{
+    // ID generator for variants.
+    return lovd_getInc('lovd_autoIncVariantID');
+}
+
+
+
+
+
+function lovd_autoIncScreeningID ()
+{
+    // ID generator for screenings.
+    return lovd_getInc('lovd_autoIncScreeningID');
+}
+
+
+
+
+
+function lovd_callJSONService ($sURL)
+{
+    // Call $sURL using lovd_php_file() and return the decoded JSON output.
+
+    $sResponse = @join('', lovd_php_file($sURL));
+    if ($sResponse) {
+        return json_decode($sResponse);
+    }
+    return false;
+}
+
+
+
+
+
+function lovd_convertCuratorID ($nLOVD2UserID)
+{
+    // Returns curator ID for given LOVD2 user ID. Return value is based on
+    // settings for fixed (default) user ID and ID translation table, both
+    // are defined in the upload form.
+    global $sFixedCuratorID, $aCuratorTranslationTable, $_WARNINGS;
+
+    $LOVD2UserIDClean = intval($nLOVD2UserID);
+    // Convert curator ID.
+    if ($LOVD2UserIDClean === 0) {
+        // '0' in LOVD2 export means the submitter ID should be used.
+        // Return false.
+        return false;
+    }
```

Comment:
In 86895d8, I've avoided the use of `intval()` so that there are no `0`s introduced there. If the input for a user ID field is `0`, this function returns `false`, downstream code should convert that based on the submitter ID.

Note that in the 86895d8 version, when e.g. input field `ID_variant_created_by_` is empty, it will be untouched by this function. However, since it is linked to the vog `created_by` field, `lovd_getRecordForHeaders()` will set that field to `0` (the default value defined in `$aFieldLinks`). It will then **not** be converted again based on the submitter ID. I think this is correct behavior.